gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -292,55 +293,94 @@ public JsonReader(Reader in) {
+   * {@linkplain Strictness#LENIENT lenient} mode, all sort of non-spec compliant JSON is accepted (see below).
+   * In {@linkplain Strictness#DEFAULT default} mode, only JSON in accordance with <a
+   * href="http://www.ietf.org/rfc/rfc4627.txt">RFC 4627</a> is accepted, with a few exceptions denoted below
+   * for backwards compatibility reasons.</p>
+   *
+   * <ul>
+   *     <li>
+   *         {@link Strictness#LENIENT}: In lenient mode, the following syntax erros are accepted:
+   *         <ul>
+   *             <li>Streams that start with the <a href="#nonexecuteprefix">non-execute prefix</a>, <code>")]}'\n"
+   *                 </code>.
+   *             <li>Streams that include multiple top-level values. With default or strict parsing,
+   *                 each stream must contain exactly one top-level value.
+   *             <li>Numbers may be {@link Double#isNaN() NaNs} or {@link Double#isInfinite() infinities}.
+   *             <li>End of line comments starting with {@code //} or {@code #} and ending with a newline character.
+   *             <li>C-style comments starting with {@code /*} and ending with
+   *                 {@code *}{@code /}. Such comments may not be nested.
+   *             <li>Names that are unquoted or {@code 'single quoted'}.
+   *             <li>Strings that are unquoted or {@code 'single quoted'}.
+   *             <li>Array elements separated by {@code ;} instead of {@code ,}.
+   *             <li>Unnecessary array separators. These are interpreted as if null
+   *                 was the omitted value.
+   *             <li>Names and values separated by {@code =} or {@code =>} instead of
+   *                 {@code :}.
+   *             <li>Name/value pairs separated by {@code ;} instead of {@code ,}.
+   *         </ul>
+   *     </li>
+   *     <li>
+   *         {@link Strictness#STRICT}: In strict mode, the only <a href="http://www.ietf.org/rfc/rfc4627.txt">RFC 4627
+   *         </a> is parsed. This means that the following syntax errors that are allowed in default and lenient mode
+   *         are not allowed:
+   *         <ul>
+   *             <li>JsonReader allows the literals {@code true}, {@code false} and {@code null}
+   *                 to have any capitalization, for example {@code fAlSe}
+   *             <li>JsonReader supports the escape sequence {@code \'}, representing a {@code '}
+   *             <li>JsonReader supports the escape sequence <code>\<i>LF</i></code> (with {@code LF}
+   *                 being the Unicode character U+000A), resulting in a {@code LF} within the
+   *                 read JSON string
+   *             <li>JsonReader allows unescaped control characters (U+0000 through U+001F)
+   *         </ul>
+   *     </li>
+   *     <li>
+   *         {@link Strictness#DEFAULT}: In default mode, the syntax errors allowed in leninet mode are not allowed.
```

Comment:
I think it might be easier to follow if the presentation were ordered on the lines of "STRICT mode accepts only the spec; DEFAULT mode is like STRICT mode but also accepts [these four things]; LENIENT mode is like DEFAULT mode but also accepts [this horrifying list of random crap]."

Reply:
I also wasn't super happy with the way I had originally written it. I took this suggestion and I think it's much clearer now.